cmd/samples/samples.go

```diff
@@ -100,40 +146,92 @@ files related to deployment of their target services.`,
 				return errors.Wrap(err, "creating sample config files")
 			}
 			printf("Config files successfully generated.")
-			if s.template != "native" {
+			if strings.Contains(s.template, "istio") {
 				printf("Please enable istio sidecar injection on the default namespace before running kubectl apply on the directory with config files.")
 			}
 			return nil
 		},
 	}
 
 	c.Flags().StringVarP(&s.ConfigPath, "config", "c", "", "path to Apigee Remote Service config file")
-	c.Flags().StringVarP(&s.template, "template", "t", "istio-1.6", "template name (options are istio-1.6, istio-1.7, native)")
+	c.Flags().StringVarP(&s.template, "template", "t", "istio-1.6", "template name (run \"samples templates\" to see available options)")
 	c.Flags().BoolVarP(&s.overwrite, "force", "f", false, "force overwriting existing directory")
 	c.Flags().StringVarP(&s.outDir, "out", "", "./samples", "directory to create config files within")
 	c.Flags().StringVarP(&s.TargetService.Name, "name", "n", "httpbin", "target service name")
-	c.Flags().StringVarP(&s.TargetService.Host, "host", "", "httpbin.org", "target service host")
-	c.Flags().StringVarP(&s.TLS.Dir, "tls", "", "", "directory for tls key and crt")
-	c.Flags().StringVarP(&s.ImageTag, "tag", "", getTagFromBuildVersion(), "version tag of the Envoy Adapter image")
+	c.Flags().StringVarP(&s.TargetService.Host, "host", "", "", "target service host (envoy templates only)")
+	c.Flags().StringVarP(&s.AdapterHost, "adapter-host", "", "", "adapter host name (envoy templates only)")
+	c.Flags().StringVarP(&s.TLS.Dir, "tls", "", "", "directory containing tls.key and tls.crt used for the adapter service (envoy templates only)")
+	c.Flags().StringVarP(&s.ImageTag, "tag", "", "", "version tag of the Envoy Adapter image (istio templates only)")
 
 	_ = c.MarkFlagRequired("config")
 
 	return c
 }
 
+func (s *samples) validateFieldsFromFlags() error {
+	if _, ok := supportedTemplates[s.template]; !ok {
```

Comment:
See comment on using map above so this function can be simplified.

Reply:
Good idea!